### ISSUE TITLE
Do I need to have a url/posts.json or am I able to retrieve data with url/posts?

### ISSUE THREAD

**Issue by oDinZu** [brockfanning/jekyll-get-json#10]
I am seeking to consume an **http://localhost:1337/posts** endpoint that displays all JSON data made public for posts. 
I created a **posts.json file in _data** directory and in my _config.yml, I added the following lines of code to GET or consume the public api. 

```
# Consume JSON from external URL
jekyll_get_json:
  - data: posts
    json: 'http://127.0.0.1:1337/posts
```

I haven't got any feedback about the plugin via logs. The plugin was added to the gemfile and config like so:
```
plugins:
  - jekyll-feed #manually generated rss.xml https://jekyllrb.com/tutorials/convert-site-to-jekyll/#10-rss-feed
  - jekyll-get-json
```
```
group :jekyll_plugins do
   gem "jekyll-feed", "~> 0.12"
   gem "jekyll-get-json"
```
On manually going through web browser for the JSON data, I retrieve the following, but the plugin doesn't seem to download or GET the information from the URL. 

on journals.html, I added the following:
```
{{ site.data.posts.title }}
```

Do I need to have a file.json or am I able to retrieve data with url/posts?
```
0 |  
id | 1
Title | "Title of Post"
Content | "Yay! I am a post><"
published_at | "2021-07-30T22:40:38.783Z"
created_at | "2021-07-30T22:40:11.170Z"
updated_at | "2021-07-30T22:40:38.798Z"
1 |  
id | 2
Title | "Another Post"
Content | "**Another post with markup**"
published_at | "2021-07-31T21:30:29.191Z"
created_at | "2021-07-31T21:30:26.836Z"
updated_at | "2021-07-31T21:30:29.207Z"
```

**Comment by csharpee** [comment]
Update: 

I tested the URL with .json file link on the web, but it didn't do anything or retrieve the data from the data.json file. 

What is it I am doing wrong? 

I am using Jekyll 4.2.0

**Comment by oDinZu** [comment]
Yeah, I have it working with a posts.json file from an external link, but it doesn't work with url/posts without .json.

**Comment by brockfanning** [comment]
I think I'm not understanding the goal entirely, but if the json file is sitting in _data, you don't need this plugin. Jekyll can parse those files all by itself.

**Comment by oDinZu** [comment]
I need to consume an api at http://localhost:1337 (Backend CMS) that gives permissions and roles and aids in management of posts and other things.

On Jekyll side, I need to pull data into _data to cache the public accessible data. To then be parsed by Jekyll site generator to build static awesome sauce. 

When using this plugin, I was unable to consume http://localhost:1337 ,but was able to consume via http://localhost:1337/post.json. My question is because of this, I wanted to make sure it was the plugin and not me. I found another plugin inspired by jekyll-get-json called jekyll-get-books https://github.com/c-leitner/jekyll-get-books and it is closely the same thing looking at the code, but that plugin seems to access http://localhost:1337 without the need for the .json extension for the URL, but it is specifically built for CSV and making use of ISBN as a reference point to download data. 

In conclusion,
When I goto http://localhost:1337/posts , I retrieve all JSON posts that are made public via CMS permissions. 
This plugin didn't allow me to consume that data without the .json extension; is that true?

**Comment by brockfanning** [comment]
There's nothing in this plugin related to file extensions, but maybe it is not sending the type of HTTP request that your /posts endpoint is expecting. This plugin is very short so you can take a look at the code. Here is the line that fetches the JSON: https://github.com/brockfanning/jekyll-get-json/blob/master/lib/jekyll-get-json/converter.rb#L25

**Comment by oDinZu** [comment]
Ok. 

Yeah, I have never programmed with Ruby before. I am reading the code, but to understand that one line is to understand how the other functions are used. 

The url where all the JSON is at endpoint: http://localhost/1337/posts
For some reason, I am unable to retrieve data from that endpoint with jekyll-get-json. 

Do I need to have the json url with the file.json extension? Or would this url work: http://localhost/1337/posts? I haven't got it to work without the .json.

**Comment by oDinZu** [comment]
When you speak of site.data, I am thinking of accessing jekyll data files in _data directory. 
I did a curl http://localhost/1337/posts and json data was retrieved. I don't know why it isn't working for the plugin? 

```
jekyll_get_json:
  - data: foo
    json: 'http://localhost/1337/posts'
```

Can data: foo be any name? Does this name create a file in _data directory? 

Then as an example: I named the data foo, then added URL (endpoint), then added to layout.html page:

`{{ site.data.foo.title }}`

the actual json data is like the following: 
```
[{"id":1,"Content":"This is the content of a p**ost!**","Title":"This is a title of a post","published_at":"2021-08-01T20:43:33.756Z","created_at":"2021-08-01T20:43:30.980Z","updated_at":"2021-08-01T20:43:33.771Z"}]
```

**Comment by oDinZu** [comment]
ok, another update. 

I got it to work with this url, but for some reason consuming my api locally doesn't work.

http://www.googleapis.com/books/v1/volumes?q=isbn

:shrug: 
I been trying to figure this out for past 4 days now and haven't moved forward.

**Comment by brockfanning** [comment]
I'm not sure either, but it is probably something about Ruby's `URI.open` that doesn't like your local web server or that local URL. It should not care whether there is an extension or not. If you want to investigate maybe you could make a quick test file:

"testing.rb"

```
require 'open-uri'
require 'json'
foo = URI.open('http://localhost:1337/posts')
puts foo
data = JSON.load(foo)
puts data
```

And then run it with Ruby: `ruby testing.rb`

Maybe it will show some exception or give some clue.
